pytorch_lightning/metrics/functional/classification.py

```diff
@@ -433,18 +430,19 @@ def recall(
         >>> x = torch.tensor([0, 1, 2, 3])
         >>> y = torch.tensor([0, 1, 2, 2])
         >>> recall(x, y)
-        tensor(0.6250)
+        tensor(0.7500)
     """
+    assert class_reduction in ('micro', 'macro', 'weighted', 'none')
```

Comment:
isn't this check already there in `class_reduce`, raised as ValueError?

Reply:
I removed these checks in the functional backend as you stated it is already checked in `class_reduce`. However, I kept them in the class backend such that an error is already produced at class initialization and not when the metric is called.